src/main/java/org/sonarqube/gradle/SonarUtils.java

```diff
@@ -41,8 +43,26 @@ static String capitalize(final String word) {
     return Character.toUpperCase(word.charAt(0)) + word.substring(1);
   }
 
-  static boolean isRoot(Project project) {
-    return project.getRootProject().equals(project);
+  static String findProjectBaseDir(Map<String, Object> properties) {
+    Path rootBaseDir = Paths.get(properties.get("sonar.projectBaseDir").toString()).toAbsolutePath().normalize();
+
+    List<Path> allProjectsBaseDir = properties.entrySet().stream()
+      .filter(e -> e.getKey().endsWith(".projectBaseDir"))
+      .map(e -> Paths.get(e.getValue().toString()))
+      .collect(Collectors.toList());
+
+    for (Path baseDir : allProjectsBaseDir) {
+      if (!baseDir.toAbsolutePath().normalize().startsWith(rootBaseDir)) {
+        int c1 = baseDir.getNameCount();
+        int c2 = rootBaseDir.getNameCount();
+        Path newBaseDir = rootBaseDir.getRoot();
+        for (int i = 0; i < c1 && i < c2 && baseDir.getName(i).equals(rootBaseDir.getName(i)); i++) {
+          newBaseDir = newBaseDir.resolve(rootBaseDir.getName(i));
+        }
+        rootBaseDir = newBaseDir;
```

Comment:
The downside of this strategy is that it will end up returning `rootBaseDir.getRoot()` if the two paths do not share the same root. Maybe you could ignore all baseDir that are not under the same root than rootBaseDir.getRoot()

Reply:
See my suggestion above